src/ios/BrowserCacheManager.swift

```diff
@@ -0,0 +1,75 @@
+import WebKit
+
+protocol WebsiteCacheManager {
+    func deleteCache(completed: (() -> ())?)
+    func deleteSessionCache(completed: (() -> ())?)
+    func clearData()
+}
+
+extension WebsiteCacheManager {
+    
+    func deleteCache(completed: (() -> ())? = nil) {
+        deleteCache(completed: completed)
+    }
+    
+    func deleteSessionCache(completed: (() -> ())? = nil) {
+        deleteSessionCache(completed: completed)
+    }
+}
+
+
+struct BrowserCacheManager: WebsiteCacheManager {
+    
+    let dataStore: WKWebsiteDataStore
+    
+    func deleteCache(completed: (() -> ())?) {
+        if #available(iOS 11.0, *), #available(OSX 10.13, *) {
+            let cookieStore = dataStore.httpCookieStore
+            cookieStore.getAllCookies {
+                let count = $0.count
+                var completedCount = 0
+                $0.forEach { cookieStore.delete($0) {
+                    completedCount += 1
+                    if completedCount == count { completed?() }
+                }}
+            }
+        } else {
+            dataStore.fetchDataRecords(ofTypes: WKWebsiteDataStore.allWebsiteDataTypes()) {
+                $0.forEach {
+                    let dataTypes = $0.dataTypes
+                    if dataTypes.contains(WKWebsiteDataTypeCookies) {
+                        self.dataStore.removeData(ofTypes: dataTypes, for: [$0]) {}
+                    }
+                }
+                completed?()
+            }
+        }
+    }
+    
+    func deleteSessionCache(completed: (() -> ())?) {
+        if #available(iOS 11.0, *), #available(OSX 10.13, *) {
+            let cookieStore = dataStore.httpCookieStore
```

Comment:
what's the impact for the WebView running in the native shell?

Reply:
That is a very good question. From this I can only say its the swift code of the implementation already made in Obj-C. So if it affects it doesn't matter and was never reported or its a known issue, one that I didn't find.